src/main/java/com/redhat/devtools/lsp4ij/LSPFileListener.java

```diff
@@ -69,15 +69,22 @@ public void fileClosed(@NotNull FileEditorManager source, @NotNull VirtualFile f
     @Override
     public void propertyChanged(@NotNull VirtualFilePropertyEvent event) {
         if (event.getPropertyName().equals(VirtualFile.PROP_NAME) && event.getOldValue() instanceof String) {
+            String oldValue = (String) event.getOldValue();

```

Comment:
To simplify code, I will write:

```java
if (event.getPropertyName().equals(VirtualFile.PROP_NAME) && 
                event.getOldValue() instanceof String oldValue &&
                event.getNewValue() instanceof String newValue) {
```

Reply:
I will move this code in a new static private method to write:

....java
public void propertyChanged(@NotNull VirtualFilePropertyEvent event) {
  if (!isRenameFile(event)) {
    return;
  }
  ....
}

private static boolean isRenameFile(@NotNull VirtualFilePropertyEvent event) {
  if (event.getPropertyName().equals(VirtualFile.PROP_NAME) && 
                event.getOldValue() instanceof String oldValue &&
                event.getNewValue() instanceof String newValue) {
    return Objects.equals(oldValue, newValue);
  }
  return false;
}
....